tests/v3_api/test_ad_custom_filter.py

```diff
@@ -0,0 +1,126 @@
+from common import *   # NOQA
+
+import requests
+
+AUTH_PROVIDER = os.environ.get('RANCHER_AUTH_PROVIDER', "")
+
+'''
+Prerequisite:
+Enable AD without TLS, and using testuser1 as admin user.
+'''
+
+# Config Fields
+HOSTNAME_OR_IP_ADDRESS = os.environ.get("RANCHER_HOSTNAME_OR_IP_ADDRESS")
+PORT = os.environ.get("RANCHER_PORT")
+CONNECTION_TIMEOUT = os.environ.get("RANCHER_CONNECTION_TIMEOUT")
+SERVICE_ACCOUNT_NAME = os.environ.get("RANCHER_SERVICE_ACCOUNT_NAME")
+SERVICE_ACCOUNT_PASSWORD = os.environ.get("RANCHER_SERVICE_ACCOUNT_PASSWORD")
+DEFAULT_LOGIN_DOMAIN = os.environ.get("RANCHER_DEFAULT_LOGIN_DOMAIN")
+USER_SEARCH_BASE = os.environ.get("RANCHER_USER_SEARCH_BASE")
+GROUP_SEARCH_BASE = os.environ.get("RANCHER_GROUP_SEARCH_BASE")
+PASSWORD = os.environ.get('RANCHER_USER_PASSWORD', "")
+
+CATTLE_AUTH_URL = \
+    CATTLE_TEST_URL + \
+    "/v3-public/"+AUTH_PROVIDER+"Providers/" + \
+    AUTH_PROVIDER.lower()+"?action=login"
+
+CATTLE_AUTH_PROVIDER_URL = \
+    CATTLE_TEST_URL + "/v3/"+AUTH_PROVIDER+"Configs/"+AUTH_PROVIDER.lower()
+
+CATTLE_AUTH_PRINCIPAL_URL = CATTLE_TEST_URL + "/v3/principals?action=search"
+
+CATTLE_AUTH_ENABLE_URL = CATTLE_AUTH_PROVIDER_URL + "?action=testAndApply"
+
+CATTLE_AUTH_DISABLE_URL = CATTLE_AUTH_PROVIDER_URL + "?action=disable"
+
+
```

Comment:
Can we add a description of the test in comments here, to give an ides about what is the test result showing and how it is affected due to setting the custom search filters